src/code.cloudfoundry.org/gorouter/route/pool.go

```diff
@@ -221,6 +241,16 @@ func NewEndpoint(opts *EndpointOpts) *Endpoint {
 		UpdatedAt:              opts.UpdatedAt,
 		LoadBalancingAlgorithm: opts.LoadBalancingAlgorithm,
 	}
+
+	// TODO: Log debug? warning when HashHeader is set but LoadBalancingAlgorithm is not LOAD_BALANCE_HB?
```

Comment:
Well, this would be a bug, no? The idea is that these cases are prevented by CC. So IMO we could add a warning level log that says something like:

> BUG: Setting lb-algo to hash should not be possible without a header, please report this to the routing-release GitHub project.

Reply:
Yep, it would be a bug. But this check was in the wrong place anyways.
We're checking if the loadbalancing algorithm is valid when writing it [from the endpoint to the endpoint pool](https://github.com/cloudfoundry/routing-release/blob/c9cbf056a28140948a50d468a02e3547dd38db89/src/code.cloudfoundry.org/gorouter/route/pool.go#L578-L591).
This will be included in a followup PR. (endpoint pool isn't in scope for the current PR)